src/MassTransit/Pipeline/Pipes/ResponsePipe.cs

```diff
@@ -72,6 +73,7 @@ public Task Send(SendContext<T> context)
         {
             context.RequestId = _context.RequestId;
             context.SourceAddress = _context.ReceiveContext.InputAddress;
+            context.TimeToLive = _context.ExpirationTime - DateTime.UtcNow;

```

Comment:
Well, this is a nullable field, and you aren't treating it that way, so this seems like it needs more work.

Reply:
it works just well, If u subtract from `null` it will return `null` as well

Reply:
https://dotnetfiddle.net/TJzzBs

Reply:
Wel, I think ExpirationTime is in UTC, and you aren't using UTC.

Reply:
` _context.ExpirationTime - DateTime.UtcNow;` I do not get it. I use UTC

Reply:
Oh, right, I was looking at the fiddle.